pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -192,19 +174,28 @@ type etcd struct {
 	role                    string
 	class                   Class
 	annotations             map[string]string
+	failureToleranceType    *gardencorev1beta1.FailureToleranceType
 	replicas                *int32
 	storageCapacity         string
 	defragmentationSchedule *string
-	nodeSpread              bool
-	zoneSpread              bool
+	caRotationPhase         gardencorev1beta1.ShootCredentialsRotationPhase
 	k8sVersion              string
+	etcd                    *druidv1alpha1.Etcd
+	backupConfig            *BackupConfig
+	hvpaConfig              *HVPAConfig
+	ownerCheckConfig        *OwnerCheckConfig
+}
+
+func (e *etcd) isFailureToleranceTypeZone() bool {
+	return e.failureToleranceType != nil && *e.failureToleranceType == gardencorev1beta1.FailureToleranceTypeZone
+}
 
-	etcd *druidv1alpha1.Etcd
+func (e *etcd) isFailureToleranceTypeNode() bool {
+	return e.failureToleranceType != nil && *e.failureToleranceType == gardencorev1beta1.FailureToleranceTypeNode
+}
```

Comment:
These functions could be general utilities which are used by other components as well. WDYT about adding this to `pkg/apis/core/v1beta1/helper`?

Reply:
The thought initially was that this is currently the only place where it is used. If we find that this will be used elsewhere we can move it. As this was specific to etcd i kept it here.

Reply:
Ok. Currently it is a method defined on etcd struct which uses the field inside the struct. If i move it to `helper` then i will have to pass `failureToleranceType`. i will make the change.

Reply:
Done